src/components/AddItem.js

```diff
@@ -1,19 +1,32 @@
 import React, { useState } from 'react';
 import { writeToFirestore } from '../lib/firebase';
 
-const AddItem = ({token}) => {
+const AddItem = ({ token, results }) => {
   let [name, setName] = useState('');
   let [frequency, setFrequency] = useState(7);
 
   function handleSubmitForm(event) {
     event.preventDefault();
     if (name.length > 0) {
-      writeToFirestore(token, {
-        name,
-        frequency,
-        lastPurchaseDate: null,
+      let count = 0;
+      results.forEach(result => {
+        if (
+          name.replace(/[\W_]/gi, '').toLowerCase() ===
+          result.name.replace(/[\W_]/gi, '').toLowerCase()
+        ) {
+          count++;
+        }
```

Comment:
We weren't sure about how to best clean this function up, as it looks a little messier than we'd like with the duplicate regex/toLowerCase. Feedback very welcome!

Reply:
Awesome job!  It's very satisfying to see it catch !@ba@nanas!  😄 

- I don't think you need the first if statement `if (name.length > 0)` because the `required` attribute is already checking if the input field is empty 
- Instead of using `forEach`, try `filter`.  You can use that in the if statement `if (results.filter(r => {}).length >0)` and then you won't have to use count
- A general bug that I noticed (not from this task), is that it currently accepts a single space as a valid entry - we might want to include some validation for that in future commits

Reply:
As for the duplicate regex code - it doesn't really bother me 😆 , but you could always make a helper function and then just do `helperFunction(name) === helperFunction(result.name)`.

Reply:
Thanks for flagging the single space as a valid submission! And I really like suggestion of filter to avoid the count.

Reply:
Dang. I was trying to figure out how to use .filter in there.

Reply:
I pushed the updates you mentioned, and have solved the bug re: whitespace only submissions of any length.